### ISSUE TITLE
Feature Request: Select Active Folder for Build and Test Tasks

### ISSUE THREAD

**Issue by cvanbeek13** [microsoft/vscode-cmake-tools#3256]
### Brief Issue Summary

I have a multi-root `code-workspace` file with my root project folder and a tests folder.  I have separate projects because the tests run locally on my windows machine with a msys64 toolchain, but the project itself is built for an ARM microcontroller with the arm-none-eabi toolchain.

I want to create separate tasks to build my project for arm, and to run my unit tests on windows.  So far, I'm focusing on running the tests.

I want to note that to do what I want to do manually, I can make sure my tests folder is selected (`++ctrl+shift+p++` `CMake: Select Active Folder`, select `Tests`) and then run the `++ctrl+shift+p++` `CMake: Run Tests` command.  When I do this, it properly builds my test app and executes the unit tests.  The issue is that I manually need to switch back to the project folder to run the build, and if I want to run the tests again, I need to manually switch back to the tests folder.  I would like to automatically do that by using tasks to switch the folder and run the build or tests.

## Attempt Number 1 - Using `cmake` Task Type

The first thing I tried was to find a way to adjust the `cmake` type test to use the tests folder.  Here is my `.code-workspace` file:

```json
{
	"folders": [
		{
			"name": "Project",
			"path": ".."
		},
		{
			"name": "Tests",
			"path": "../tests"
		}
	],
	"tasks": {
		"version": "2.0.0",
		"tasks": [
                         {
				"type": "cmake",
				"command": "test",
				"options": {
					"cwd": "${workspaceFolder}/tests"
				},
				"group": {
					"kind": "test",
					"isDefault": true
				},
				"problemMatcher": [
					"$gcc"
				],
			}
                 ]
         }
}
```

Adjusting the `cwd` parameter had no effect though, and the task always executed on the `Project` folder.  I couldn't find another relevant parameter to try to change.

## Attempt Number 2 - Using Task to Switch Folders

My second idea was to switch folders using another task which is called as a dependency of my test task.  Here is my configuration to do that:

```json
{
	"folders": [
		{
			"name": "Project",
			"path": ".."
		},
		{
			"name": "Tests",
			"path": "../tests"
		}
	],
	"tasks": {
		"version": "2.0.0",
		"tasks": [
			{
				"label": "SwitchToTests",
				"type": "process",
				"command": "${input:switchToTests}",
				"problemMatcher": []
			},
			{
				"label": "CMake Build Tests",
				"type": "cmake",
				"command": "build",
				"dependsOn": [
					"SwitchToTests"
				],
				"dependsOrder": "sequence",
				"problemMatcher": [
					"$gcc"
				]
			},
			{
				"label": "CMake Test",
				"type": "cmake",
				"command": "test",
				"options": {
					"cwd": "${workspaceFolder}/tests"
				},
				"group": {
					"kind": "test",
					"isDefault": true
				},
				"problemMatcher": [
					"$gcc"
				],
				"dependsOn": [
					"CMake Build Tests"
				],
				"dependsOrder": "sequence",
			}
		],
		"inputs": [
			{
				"id": "switchToTests",
				"type": "command",
				"command": "cmake.selectActiveFolder",
				"args": [
					"tests"
				]
			},
		],
         }
}
```

This almost works, but it always opens up a input window to select with folder to select when the `cmake.selectActiveFolder` command runs:

![image](https://github.com/microsoft/vscode-cmake-tools/assets/20888380/cd3b0552-e8ea-4392-995a-0c6b8defb0c7)

Is there any way to get VSCode to automatically select the `tests` folder?  I couldn't get this working with the `args` parameter.  Otherwise, is there another way to accomplish what I'm trying to do?

**Comment by benmcmorran** [comment]
Thanks for the suggestion! I'm not aware of a way to do this right now, but it seems reasonable to allow specifying a folder as a parameter to the `cmake.selectActiveFolder` command. We likely won't be able to get to this work immediately, but if someone else is interested in contributing a PR, the change needs to happen [here in the source code](https://github.com/microsoft/vscode-cmake-tools/blob/2014508a34d6cb916b47434585538a568d7d678d/src/extension.ts#L544-L558).

**Comment by cvanbeek13** [comment]
I added a PR #3258

**Comment by benmcmorran** [comment]
Fixed by #3528
